doc/spec/json.md

```diff
@@ -0,0 +1,76 @@
+# Docker Distribution JSON Canonicalization
+
+## Introduction
+
+To provide consistent content hashing of JSON objects throughout Docker
+Distribution APIs, a canonical JSON format has been defined. Adopting such a
+canonicalization also aids in caching JSON responses.
+
+## Rules
+
+Compliant JSON should conform to the following rules:
+
+1. All generated JSON should comply with [RFC
+   7159](http://www.ietf.org/rfc/rfc7159.txt).
+2. Resulting "JSON text" shall always be encoded in UTF-8.
+3. Unless a canonical key order is defined for a particular schema, object
```

Comment:
Legacy code or working with libraries that already generate JSON in a specific order (like TUF, which seems to have a consistent, non-lexical ordering).